libcudacxx/include/cuda/std/__cccl/extended_data_types.h

```diff
@@ -90,7 +90,7 @@
 
 #if !defined(CCCL_DISABLE_FLOAT128_SUPPORT) && _CCCL_OS(LINUX) && !_CCCL_ARCH(ARM64)
 #  if (defined(__CUDACC_RTC_FLOAT128__) || defined(__SIZEOF_FLOAT128__) || defined(__FLOAT128__)) /*HOST COMPILERS*/
-#    if _CCCL_CUDA_COMPILER(NVHPC) \
+#    if _CCCL_HOST_COMPILATION() || _CCCL_CUDA_COMPILER(NVHPC) \
```

Comment:
I believe we should split  this into `_CCCL_HOST_COMPILATION()` and `_CCCL_DEVICE_COMPILATION`